pkg/controller/master/image/backing_image_controller.go

```diff
@@ -0,0 +1,73 @@
+package image
+
+import (
+	"reflect"
+
+	"github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta1"
+	"github.com/longhorn/longhorn-manager/types"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	v1beta12 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	harvesterv1beta1 "github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+	lhv1beta1 "github.com/harvester/harvester/pkg/generated/controllers/longhorn.io/v1beta1"
+	"github.com/harvester/harvester/pkg/ref"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+// backingImageHandler syncs upload progress from backing image to vm image status
+type backingImageHandler struct {
+	vmImages          harvesterv1beta1.VirtualMachineImageClient
+	vmImageCache      harvesterv1beta1.VirtualMachineImageCache
+	backingImages     lhv1beta1.BackingImageClient
+	backingImageCache lhv1beta1.BackingImageCache
+}
+
+func (h *backingImageHandler) OnChanged(_ string, backingImage *v1beta1.BackingImage) (*v1beta1.BackingImage, error) {
+	if backingImage == nil || backingImage.DeletionTimestamp != nil {
+		return nil, nil
+	}
+	if backingImage.Spec.SourceType != types.BackingImageDataSourceTypeUpload || backingImage.Annotations[util.AnnotationImageID] == "" {
+		return nil, nil
+	}
```

Comment:
since the LH backing image also tracked its progress for URL data source type, please add to sync its status instead of making it an instant complete as before.

Reply:
It is not a 1-1 mapping, LH backing image records progress for multiple disk files. The download progress varies on different nodes.

Reply:
understand, as long as we can use the first complete node as the default tracking progress, it will make the status look much solid when compared with the instant completion, WDUT.

Reply:
Updated to sync progress for download type. Changed the Uploaded condition to Imported to fit both types.